### PR TITLE
remove ParseForeachArgs template

### PR DESCRIPTION
**Pull request by WalterBright**
It doesn't add any value.

### CODE REVIEW

**Comment by WalterBright** [comment]
FreeBSD 11.4 x64, DMD (bootstrap) Failing after 11m — Task Summary
```
make[1]: *** [posix.mak:412: unittest/std/concurrency.run] Segmentation fault (core dumped)
```
The Phobos concurrency code ought to be more robust.

**Comment by WalterBright** [comment]
The FreeBSD failure is a well-trod heisenbug in std.concurrency and can be ignored.

**Comment by PetarKirov** [comment]
> The FreeBSD failure is a well-trod heisenbug in std.concurrency and can be ignored.

It seems so indeed - I restarted the test job and now it's green.

File: src/dmd/parse.d [hunk 1]
```diff
@@ -5401,7 +5386,7 @@ class Parser(AST) : Lexer
      * If `isStatic` is true, `isDecl` can be true to indicate that a
      * `static foreach` declaration should be parsed.
      */
-    private ParseForeachRet!(isStatic, isDecl) parseForeach(bool isStatic, bool isDecl)(Loc loc, ParseForeachArgs!(isStatic, isDecl) args)
+    private ParseForeachRet!(isStatic, isDecl) parseForeach(bool isStatic, bool isDecl)(Loc loc, AST.Dsymbol* pLastDecl)
```
Comment (PetarKirov) [inline src/dmd/parse.d:5389]: Please update the DDoc to specify what `pLastDecl` is and that if `isDecl` is `false`, then `null` should be passed to `pLastDecl`.

**Review by PetarKirov — COMMENTED** [review]
In general, I prefer the previous version as it makes the API contact more foolproof, but since this is a private function that doesn't matter as much.

File: src/dmd/parse.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/parse.d:5389]: This is covered in the followup to this PR https://github.com/dlang/dmd/pull/12922

**Review by kinke — APPROVED** [review]

**Comment by kinke** [comment]
(Merging eagerly to unblock the follow-up PR.)

**Pull request merged**